core/src/main/java/com/datastax/oss/driver/api/core/metadata/EndPoint.java

```diff
@@ -40,6 +40,17 @@ public interface EndPoint {
   @NonNull
   SocketAddress resolve();
 
+  /**
+   * Returns a possibly unresolved instance to a socket address.
+   *
+   * <p>This should be called when the address does not need to be proactively resolved. For example
+   * if the node hostname or port number is needed.
+   */
+  @NonNull
+  default SocketAddress retrieve() {
+    return resolve();
+  }
```

Comment:
I'm wondering if it makes sense to provide a default implementation for the new method.  It's unlikely we'll be adding a new endpoint type anytime soon but failing to distinguish between "look up the addresses again" and "give me the addresses you got last time you looked them up" is what got us into this situation in the first place.  I kinda feel like having this as a default might lend itself to landing back in exactly that situation.